packages/authentication/cardstack/middleware.js

```diff
@@ -143,6 +128,24 @@ class Authentication {
       user = await this._processExternalUser(result, source);
     }
 
+    if (result && result.meta && result.meta.partialSession) {
```

Comment:
If this checks user.meta instead of result.meta, it will give the userTemplate control over whether this is a partial response.

Reply:
@ef4 Yeah, but email-auth is not returning valid JSONAPI, and hence `_processExternalUser` returns null. So that is why I need to check both `result` and `user`

Reply:
I see, the only reason it's not returning valid JSON:API is because it was relying on getting `{type: "partial-sessions" }` as a default, which now happens too late. Let's jettison the idea of having a default type and just make the plugin set it. So email-auth should just add that type explicitly, and we can drop the stuff at line 140 about setting a default type.

Reply:
Sounds good